datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -320,11 +320,12 @@ public DetachRecipeV4Response detachRecipe(SdxCluster sdxCluster, DetachRecipeV4
 
     public void updateRangerRazEnabled(SdxCluster sdxCluster) {
         String initiatorUserCrn = ThreadBasedUserCrnProvider.getUserCrn();
+        DetailedEnvironmentResponse environment = environmentClientService.getByName(sdxCluster.getEnvName());
         ThreadBasedUserCrnProvider.doAsInternalActor(() -> {
             RangerRazEnabledV4Response response = stackV4Endpoint.rangerRazEnabledInternal(WORKSPACE_ID_DEFAULT, sdxCluster.getCrn(), initiatorUserCrn);
             if (response.isRangerRazEnabled()) {
                 if (!sdxCluster.isRangerRazEnabled()) {
-                    validateRazEnablement(sdxCluster.getRuntime(), response.isRangerRazEnabled(), environmentClientService.getByName(sdxCluster.getEnvName()));
+                    validateRazEnablement(sdxCluster.getRuntime(), response.isRangerRazEnabled(), environment);
```

Comment:
I am not sure we need to pass the whole DetailedEnvironmentResponse object as a parameter, only cloudPlatform and an account id is needed for the  `validateRazEnablement` method

Reply:
Yes, I hadn't completely refactored the `validateRazEnablement` method as it was invoked even while creating an `SdxCluster`

Reply:
Here is the other invocation

https://github.com/hortonworks/cloudbreak/blob/1e2fc379213a680902af5e61ba401f460b751d5e/datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java#L498-L508

Reply:
How has this slipped through testing? Could you point me to the e2e test for this feature?

Reply:
I am unsure of e2e testing for this. The original changes were made in the PR referenced in the description

Reply:
regarding e2e test: as I see the previous change which caused the bug which should be fixed by this pr, was never tested as it never could work. I would expect that this PR would contain a test, e2e or integration, which would make sure that this is working now and ensure that it works later on.

Reply:
we have several mock tests for SDX creation, like `MockSdxTests`, that can be extended and that will be executed in "PR-time"

Reply:
Okay, will update this PR with the required mockSdx tests to ensure the feature doesn't break.

Reply:
have you considered mock tests before implementing e2e test? if there is no need to test it on provider side, mock test is much cheaper and faster

Reply:
I wrote and verified e2e tests locally, as I was having some issues when getting mock tests to run locally

Reply:
ok, i am fine with this